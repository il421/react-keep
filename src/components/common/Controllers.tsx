import React from "react";
import { AlignItems, JustifyContent } from "../../common";
import { FlexBox, IconButton } from "../ui-components";
import "../../styles/components/common/_controllers.scss";
import { NoteType } from "../notes";

interface Controllers extends Partial<Pick<HTMLElement, "className">> {
  isMobile?: boolean;
  openDialog: (type: NoteType) => void;
}

export const Controllers: React.FunctionComponent<Controllers> = ({
  className,
  isMobile = false,
  openDialog,
}) => {
  const handleOnClick = (type: NoteType) => () => openDialog(type);

  return (
    <div
      className={`controllers  ${
        isMobile && "controllers--mobile"
      } ${className}`}
    >
      <FlexBox
        vertical={isMobile}
        justifyContent={JustifyContent.spaceBetween}
        alignItems={AlignItems.center}
      >
        <IconButton
          className="controllers__text"
          onButtonClick={handleOnClick(NoteType.text)}
          icon="align-left"
          size="2x"
        />

        <IconButton
          className="controllers__list"
          onButtonClick={handleOnClick(NoteType.list)}
          icon="check-square"
          size="2x"
        />

        <IconButton
          className="controllers__image"
          onButtonClick={handleOnClick(NoteType.image)}
          icon="image"
          size="2x"
        />
      </FlexBox>
    </div>
  );
};
