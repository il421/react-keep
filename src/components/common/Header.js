import React from 'react';
import { connect } from 'react-redux';

import { startLogout } from '../../actions/auth';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import Search from './Search';

export const Header = ({ startLogout, auth, showSidebar }) => {

  return (
    <header className="header" onClick={(evt) => evt.stopPropagation()}>
      <div className="content-container">
        <div className="header__content">
          <div className="header__sidebar">

            <button className="button button--sidenav" onClick={() => showSidebar()}>
              <FontAwesomeIcon icon="bars" size="2x" />
            </button>

          </div>
          <div>
            <h1 className="header__title">Keep Me</h1>
          </div>
          <div className="header__search">
            <Search />
          </div>
          <div className="header__user-box user-box">
            <div className="user-box__photo">
              <img src={ auth.url } width={ 50 } height={ 50 } />
            </div>
            <div className="user-box__name show-for-mobile">{ auth.name }</div>
            <button className="user-box__logout button button--sidenav" onClick={ startLogout }>
              <FontAwesomeIcon icon="sign-out-alt" size="lg" />
            </button>
          </div>
        </div>
      </div>
    </header>
  );
};


const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

const mapStateToProps = (state) => {
  return {
    auth: state.auth,
    url: state.url,
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Header);
