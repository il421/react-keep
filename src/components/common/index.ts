export * from "./Controllers";
export * from "./DashboardPage";
export * from "./Header";
export * from "./NotFoundPage";
export * from "./Search";
export * from "./SideBar";
