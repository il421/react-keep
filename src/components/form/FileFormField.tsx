import React from "react";
import { Field } from "react-final-form";
import { ExposedFieldProps, FieldAdapterProps } from "./BaseForm.types";
import Dropzone from "react-dropzone";

// @TODO need to complete
const FieldAdapter: React.FunctionComponent<FieldAdapterProps> = ({
  className,
  input,
  meta,
  name,
  ...rest
}) => {
  return (
    <div className={className}>
      <Dropzone {...input}
      >
        {({ getRootProps, getInputProps }) => (
          <div
            {...getRootProps()}
          >
            <input {...getInputProps()} />
            <p>hihi</p>
          </div>
        )}
      </Dropzone>
      {meta.error && meta.touched && <span>{meta.error}</span>}
    </div>
  );
};

export const FileFormField: React.FunctionComponent<ExposedFieldProps<string, HTMLElement>
  & {
  className: string,
}> = (props) => (
  <Field
    {...props}
    name={props.name}
    component={FieldAdapter}
    parse={(value: string) => value.trim() ?? ""}
  />
);
