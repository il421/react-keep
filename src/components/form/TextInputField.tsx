import React from "react";
import { Field } from "react-final-form";
import TextareaAutosize from "react-textarea-autosize";
import { ExposedFieldProps, FieldAdapterProps } from "./BaseForm.types";

const FieldAdapter: React.FunctionComponent<FieldAdapterProps> = ({
  className,
  autoFocus = false,
  input,
  meta,
  isTextArea = false,
  ...rest
}) => {
  return (
    <div className={className}>
      {isTextArea ? (
        <TextareaAutosize
          {...input}
          {...rest}
          spellCheck="false"
          autoComplete="new-password"
          autoFocus={autoFocus}
        />
      ) : (
        <input
          {...input}
          {...rest}
          autoComplete="new-password"
          autoFocus={autoFocus}
        />
      )}
      {meta.error && meta.touched && <span>{meta.error}</span>}
    </div>
  );
};

export const TextInputField: React.FunctionComponent<ExposedFieldProps<
  string,
  HTMLElement
> & {
  type?: string;
  placeholder?: string;
  className: string;
  isTextArea?: boolean;
  autoFocus?: boolean;
}> = props => (
  <Field
    {...props}
    name={props.name}
    component={FieldAdapter}
    parse={(value: string) => value ?? ""}
  />
);
