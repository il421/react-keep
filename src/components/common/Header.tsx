import React from "react";
import { connect } from "react-redux";
import { useHistory } from "react-router-dom";
import { stringify } from "query-string";
import { AuthStoreState, Store } from "../../store/store.types";
import { QueryKeys, RouteActions } from "../../routers/Routing";
import { startLogout } from "../../actions/auth";
import Search from "./Search";
import { Controllers } from "./Controllers";
import {
  IconButton,
  UserPhoto,
  ContentContainer,
  LinkButton
} from "../ui-components";
import "../../styles/components/header/_header.scss";
import "../../styles/components/header/_user-box.scss";
import "../../styles/components/common/_visibility.scss";
import { ThunkDispatch } from "redux-thunk";

interface HeaderProps {
  showSidebar: (value: boolean) => void;
}

interface StateProps {
  auth: AuthStoreState;
}

interface DispatchProps {
  startLogout: () => void;
}

type Props = HeaderProps & DispatchProps & StateProps;

const Header: React.FunctionComponent<Props> = ({
  startLogout,
  auth,
  showSidebar
}) => {
  const history = useHistory();

  const onClickHandler = (key: keyof typeof QueryKeys) => {
    const query = stringify({
      [key]: RouteActions.edit
    });
    history.push(`${history.location.pathname}?${query}`);
  };

  return (
    <header className="header" onClick={evt => evt.stopPropagation()}>
      <ContentContainer>
        <div className="header__content">
          <IconButton
            className="header__bar"
            icon="bars"
            size="2x"
            onClick={() => showSidebar(true)}
          />

          <h1 className="header__title">Keep Me</h1>

          <Search wrapperClass="header__search" />

          <Controllers className="hide-for-mobile" />

          <div className="header__user-box user-box">
            {auth.name && auth.url && (
              <UserPhoto
                className="user-box__photo"
                src={auth.url}
                height={50}
                width={50}
              />
            )}
            <LinkButton
              text={auth.name ?? "Update profile"}
              type="button"
              className="user-box__link"
              onClick={() => onClickHandler(QueryKeys.user)}
            />
            <IconButton
              className="user-box__logout"
              icon="sign-out-alt"
              size="lg"
              onClick={startLogout}
            />
          </div>
        </div>
      </ContentContainer>
    </header>
  );
};

const mapDispatchToProps = (
  dispatch: ThunkDispatch<{}, {}, any>
): DispatchProps => ({
  startLogout: () => dispatch(startLogout())
});

const mapStateToProps = (state: Store): StateProps => {
  return {
    auth: state.auth
  };
};

export default connect<StateProps, DispatchProps, HeaderProps, Store>(
  mapStateToProps,
  mapDispatchToProps
)(Header);
