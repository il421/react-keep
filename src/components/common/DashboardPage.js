import React, { useState } from 'react';

import NoteSelection from '../note/NoteSelection';
import NotesList from '../note/NotesList';
import Header from './Header';
import SideBar from './SideBar';


const DashboardPage = () => {
  let [showBar, toggleSidebarDisplay] = useState(false);

  const showSidebar = () => {
    toggleSidebarDisplay(showBar = true);
  };

  const hideSidebar = () => {
    toggleSidebarDisplay(showBar = false);
  };

  return (
    <div className="dashboard">
      <Header showSidebar={ showSidebar } />
      <NoteSelection className="content-container"/>
      <NotesList/>

      <SideBar showBar={ showBar } hideSidebar={ hideSidebar } />
    </div>
  );
};
export default DashboardPage;
