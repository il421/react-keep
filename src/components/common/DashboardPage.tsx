import React from "react";
import { ToastContainer } from "react-toastify";
import { History, Path } from "history";
import { JustifyContent } from "../../common";
import { Header } from "./Header";
import { SideBar } from "./SideBar";
import { Controllers } from "./Controllers";
import UserFormModal from "../login/UserFormModal";
import "../../styles/components/common/_dashboard.scss";
import "../../styles/components/notes/_note-modal.scss";
import { QueryKeys } from "../../routers/Routing";
import { stringify } from "query-string";
import {
  NoteType,
  ImageNoteFormModal,
  TextNoteFormModal,
  ListNoteFormModal,
} from "../notes";
import { FlexBox } from "../ui-components";
import { NotesList } from "../notes/NotesList";

export interface DashboardPageProps {
  history: History;
}

export const onNoteSelected = (options: {
  type: NoteType;
  id: string;
  pathname: string;
  push: (path: Path) => void;
}) => {
  let key: keyof typeof QueryKeys;
  const { type, id, pathname, push } = options;
  switch (type) {
    case NoteType.text:
      key = QueryKeys.text;
      break;
    case NoteType.list:
      key = QueryKeys.list;
      break;
    default:
      key = QueryKeys.image;
  }

  const query = stringify({
    [key]: id,
  });
  push(`${pathname}?${query}`);
};

export const DashboardPage: React.FunctionComponent<DashboardPageProps> = ({
  history,
}): JSX.Element => {
  return (
    <FlexBox
      vertical={true}
      justifyContent={JustifyContent.spaceBetween}
      className="dashboard"
    >
      <Header history={history} />
      <NotesList
        onNoteSelected={(type: NoteType, id: string) =>
          onNoteSelected({
            type,
            id,
            pathname: history.location.pathname,
            push: history.push,
          })
        }
      />

      <SideBar />
      <TextNoteFormModal history={history} />
      <ListNoteFormModal history={history} />
      <ImageNoteFormModal history={history} />
      <UserFormModal />
      <Controllers
        isMobile
        openDialog={(query) =>
          history.push(`${history.location.pathname}?${query}`)
        }
      />

      <ToastContainer autoClose={2000} />
    </FlexBox>
  );
};
