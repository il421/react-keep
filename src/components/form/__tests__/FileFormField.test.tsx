import React from "react";
import { mount, ReactWrapper } from "enzyme";
import { TextInputField, TextInputFieldProps } from "../TextInputField";
import { Form } from "react-final-form";
import { ValidationErrors } from "final-form";
import { Errors } from "../../../common";
import { FileFormField, FileFormFieldProps } from "../FileFormField";

let props: FileFormFieldProps, wrapper: ReactWrapper<FileFormFieldProps, {}>;

beforeEach(() => {
  props = {
    className: "className",
    name: "name",
    id: "id",
  };
});

test("should render FileFormField", () => {
  wrapper = mount<FileFormFieldProps>(
    <Form initialValues={{ name: "" }} onSubmit={() => {}}>
      {() => <FileFormField {...props} />}
    </Form>
  );
  expect(wrapper.debug()).toMatchSnapshot();
  expect(wrapper.find("input[name='name']").exists()).toBeTruthy();
});

test("should render error message after invalid submitting", () => {
  wrapper = mount<FileFormFieldProps>(
    <Form
      initialValues={{ name: undefined }}
      onSubmit={() => {}}
      validate={(values: { name: File | undefined }): ValidationErrors => {
        const errors: ValidationErrors = {};
        const { name } = values;

        if (!name) {
          errors.name = Errors.required;
        }

        return errors;
      }}
    >
      {({ handleSubmit }) => (
        <form onSubmit={handleSubmit}>
          <FileFormField {...props} />
        </form>
      )}
    </Form>
  );

  expect(wrapper.find("form").exists()).toBeTruthy();
  wrapper.find("form").simulate("focus").simulate("submit");
  expect(wrapper.find("#error-message-name").exists()).toBeTruthy();
  expect(wrapper.find("#error-message-name").text()).toBe(Errors.required);
});
