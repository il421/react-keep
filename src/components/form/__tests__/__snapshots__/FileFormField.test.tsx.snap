// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FileFormField 1`] = `
"<ReactFinalForm initialValues={{...}} onSubmit={[Function: onSubmit]}>
  <FileFormField className=\\"className\\" name=\\"name\\" id=\\"id\\">
    <Field className=\\"className\\" name=\\"name\\" id=\\"id\\" component={[Function: FieldAdapter]}>
      <FieldAdapter input={{...}} meta={{...}} className=\\"className\\" id=\\"id\\" render={[undefined]}>
        <div className=\\"className\\">
          <input render={[undefined]} name=\\"name\\" id=\\"id\\" type=\\"file\\" onChange={[Function: handleChange]} accept=\\"image/*\\" />
        </div>
      </FieldAdapter>
    </Field>
  </FileFormField>
</ReactFinalForm>"
`;
